### PR TITLE
Make it partially compatible with scala 2.11 and Xlint free and minor cleanups

### PR DESCRIPTION
**Pull request by douglaz**

### CODE REVIEW

**Comment by fparisotto** [comment]
LGTM

File: src/main/scala/ignition/core/jobs/utils/SparkContextUtils.scala [hunk 1]
```diff
@@ -289,7 +291,7 @@ object SparkContextUtils {
                        maxBytesPerPartition: Long,
                        minPartitions: Int,
                        sizeBasedFileHandling: SizeBasedFileHandling): RDD[String] = {
-      val smallPartitionedFiles = sc.parallelize(smallFiles.map(_.path).map(file => file -> ()), 2).partitionBy(createSmallFilesPartitioner(smallFiles, maxBytesPerPartition, minPartitions, sizeBasedFileHandling))
+      val smallPartitionedFiles = sc.parallelize(smallFiles.map(_.path).map(file => file -> null), 2).partitionBy(createSmallFilesPartitioner(smallFiles, maxBytesPerPartition, minPartitions, sizeBasedFileHandling))
```
Comment (sisso) [inline src/main/scala/ignition/core/jobs/utils/SparkContextUtils.scala]: No workaround for this null?

**Comment by sisso** [comment]
lgtm

If possible remove that null :P

**Comment by douglaz** [comment]
@sisso look the implementation of the countByValue on Spark. It uses the same strategy. null is the Right Way

**Pull request merged**